packages/salesforce/src/Adaptor.js

```diff
@@ -659,56 +659,103 @@ function setApiVersion(apiVersion) {
 
   return version;
 }
+
 /**
- * Creates a connection.
+ * Establish a connection to Salesforce using username and password.
  * @example
- * createConnection(state)
+ * createBasicAuthConnection(state)
  * @function
  * @param {State} state - Runtime state.
  * @returns {State}
  */
-function createConnection(state) {
-  const { loginUrl, apiVersion } = state.configuration;
+async function createBasicAuthConnection(state) {
+  console.log('Attempting to connect with Basic Auth');
+
+  const { loginUrl, apiVersion, username, password, securityToken } =
+    state.configuration;
 
-  if (!loginUrl) {
-    throw new Error('loginUrl missing from configuration.');
+  if (!loginUrl || !username || !password) {
+    throw new Error(
+      'Incomplete configuration for basic authentication. Provide loginUrl, username, password, and securityToken if required.'
+    );
   }
 
+  const connectionOptions = { loginUrl };
+
+  if (apiVersion) {
+    connectionOptions.version = setApiVersion(apiVersion);
+  }
+
+  const connection = new jsforce.Connection(connectionOptions);
+
+  console.info(`Logging in as ${username}.`);
+
+  await connection.login(
+    username,
+    securityToken ? password + securityToken : password
+  );
+  // NOTE: Uncomment this to debug connection issues.
+  // .then(response => {
+  //   console.log(connection);
+  //   console.log(response);
+  //   return state;
+  // })
+
   return {
     ...state,
-    connection: apiVersion
-      ? new jsforce.Connection({
-          loginUrl,
-          version: setApiVersion(apiVersion),
-        })
-      : new jsforce.Connection({ loginUrl }),
+    connection,
   };
 }
 
 /**
- * Performs a login.
+ * Establish a connection to Salesforce using Access Token.
```

Comment:
I don't think that this or `createBasicAuthConnection` should be documented like this. They're both private and internal functions - `createConnection` is the public interface (and even then it's an internal function, it shouldn't be documented for job writers)

Reply:
@josephjclark i have update the docstring and add `@private`, Please take a look